diff --git a/Dockerfile-linux.template b/Dockerfile-linux.template
index 75a6d2d8..3f386f0a 100644
--- a/Dockerfile-linux.template
+++ b/Dockerfile-linux.template
@@ -389,14 +389,18 @@ RUN set -eux; \
 	make -j "$(nproc)"; \
 	find -type f -name '*.a' -delete; \
 	make install; \
-	find \
-		/usr/local \
-		-type f \
-		-perm '/0111' \
-		-exec sh -euxc ' \
-			strip --strip-all "$@" || : \
-		' -- '{}' + \
-	; \
+	if ! php -n -d 'display_errors=stderr' -r 'exit(ZEND_DEBUG_BUILD ? 0 : 1);' > /dev/null; then \
+		# only "strip" binaries if we aren't using a debug build of PHP
+		# (none of our builds are debug builds, but PHP might be recompiled with "--enable-debug" configure option)
+		# https://github.com/docker-library/php/issues/1268
+		find \
+			/usr/local \
+			-type f \
+			-perm '/0111' \
+			-exec sh -euxc ' \
+				strip --strip-all "$@" || : \
+			' -- '{}' +; \
+	fi; \
 	make clean; \
 	\
 # https://github.com/docker-library/php/issues/692 (copy default example "php.ini" files somewhere easily discoverable)
