diff --git a/ext/pdo_oci/oci_statement.c b/ext/pdo_oci/oci_statement.c
index 0ab043f041d..6ab85e52bcf 100644
--- a/ext/pdo_oci/oci_statement.c
+++ b/ext/pdo_oci/oci_statement.c
@@ -646,22 +646,26 @@ static ssize_t oci_blob_read(php_stream *stream, char *buf, size_t count)
 {
 	struct oci_lob_self *self = (struct oci_lob_self*)stream->abstract;
 	ub4 amt;
-	sword r;
+    sword r;
+	oraub8 byte_amt = (oraub8) count;
+	oraub8 char_amt = 0;
 
-	amt = (ub4) count;
-	r = OCILobRead(self->E->svc, self->E->err, self->lob,
-		&amt, self->offset, buf, (ub4) count,
-		NULL, NULL, 0, SQLCS_IMPLICIT);
+	ub1 csfrm;
+	OCILobCharSetForm(self->S->H->env, self->S->err, self->lob, &csfrm);
+
+	r = OCILobRead2(self->E->svc, self->E->err, self->lob,
+		&byte_amt, &char_amt, (oraub8) self->offset, buf, (oraub8) count,
+        OCI_ONE_PIECE, NULL, NULL, 0, csfrm);
 
 	if (r != OCI_SUCCESS && r != OCI_NEED_DATA) {
 		return (size_t)-1;
 	}
 
-	self->offset += amt;
-	if (amt < count) {
+	self->offset += csfrm == 0 ? byte_amt : char_amt;
+	if (byte_amt < count) {
 		stream->eof = 1;
 	}
-	return amt;
+	return byte_amt;
 }
 
 static int oci_blob_close(php_stream *stream, int close_handle)
